Update Rcpp dependencies

## CRAN Message
>Compilation fails with _R_USE_STRICT_R_HEADERS_=true, which defines
STRICT_R_HEADERS to 1 which removes
>
>- the legacy definition of PI (use POSIX's M_PI, available in R fer ever).
>- the RS.h declarations for Calloc, Realloc, Free (use R_ forms i
  available since R 3.4.0).
> 
>The aim is to clean the namespace: in particular having a definition
for Free has conflicted with some packages' C++ code.
>
>It is planned that STRICT_R_HEADERS=1 will become the default for 4.5.0,
for which it would be good if all CRAN packages, and in particular the
ones with strong reverse dependencies, compile/install ok with the new
default.
>
>Your package is among the ones with strong reverse dependencies which
need updating.  We would thus really appreciate if you could provide a
new version of your package as soon as possible which checks ok with
STRICT_R_HEADERS=1.
>
>You can verify that your package checks ok with STRICT_R_HEADERS=1 via R
CMD check --as-cran using a current version of R-devel.

## Strict R Check Results
>* installing *source* package ‘BoutrosLab.plotting.general’ ...
** package ‘BoutrosLab.plotting.general’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
using C compiler: ‘gcc-14 (GCC) 14.2.0’
make[1]: Entering directory '/data/gannet/ripley/R/packages/tests-Strict/BoutrosLab.plotting.general/src'
gcc-14 -I"/data/gannet/ripley/R/R-devel/include" -DNDEBUG   -I/usr/local/include  -DSTRICT_R_HEADERS=1  -fpic  -g -O2 -Wall -pedantic -mtune=native -Wp,-D_FORTIFY_SOURCE=3 -fexceptions -fstack-protector-strong -fstack-clash-protection -fcf-protection -Werror=implicit-function-declaration -Wstrict-prototypes  -c distance.c -o distance.o
gcc-14 -I"/data/gannet/ripley/R/R-devel/include" -DNDEBUG   -I/usr/local/include  -DSTRICT_R_HEADERS=1  -fpic  -g -O2 -Wall -pedantic -mtune=native -Wp,-D_FORTIFY_SOURCE=3 -fexceptions -fstack-protector-strong -fstack-clash-protection -fcf-protection -Werror=implicit-function-declaration -Wstrict-prototypes  -c init.c -o init.o
gcc-14 -I"/data/gannet/ripley/R/R-devel/include" -DNDEBUG   -I/usr/local/include  -DSTRICT_R_HEADERS=1  -fpic  -g -O2 -Wall -pedantic -mtune=native -Wp,-D_FORTIFY_SOURCE=3 -fexceptions -fstack-protector-strong -fstack-clash-protection -fcf-protection -Werror=implicit-function-declaration -Wstrict-prototypes  -c ks.c -o ks.o
ks.c: In function 'K':
ks.c:151:18: error: implicit declaration of function 'Calloc'; did you mean 'calloc'? [-Wimplicit-function-declaration]
  151 |    H = (double*) Calloc(m * m, double);
      |                  ^~~~~~
      |                  calloc
ks.c:151:32: error: expected expression before 'double'
  151 |    H = (double*) Calloc(m * m, double);
      |                                ^~~~~~
ks.c:152:32: error: expected expression before 'double'
  152 |    Q = (double*) Calloc(m * m, double);
      |                                ^~~~~~
ks.c:180:4: error: implicit declaration of function 'Free'; did you mean 'free'? [-Wimplicit-function-declaration]
  180 |    Free(H);
      |    ^~~~
      |    free
ks.c: In function 'm_power':
ks.c:218:33: error: expected expression before 'double'
  218 |     B = (double*) Calloc(m * m, double);
      |                                 ^~~~~~
make[1]: *** [/data/gannet/ripley/R/R-devel/etc/Makeconf:195: ks.o] Error 1
make[1]: Leaving directory '/data/gannet/ripley/R/packages/tests-Strict/BoutrosLab.plotting.general/src'
ERROR: compilation failed for package ‘BoutrosLab.plotting.general’
>* removing ‘/data/gannet/ripley/R/packages/tests-strict/Libs/BoutrosLab.plotting.general-lib/BoutrosLab.plotting.general’
Command exited with non-zero status 1
Time 0:02.89, 1.21 + 0.28

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -2,7 +2,7 @@ Package: BoutrosLab.plotting.general
 Version: 7.1.2
 Type: Package
 Title: Functions to Create Publication-Quality Plots
-Date: 2024-04-02
+Date: 2024-09-06
 Authors@R: c(person("Paul Boutros", role = c("aut", "cre"), email = "PBoutros@mednet.ucla.edu"),
              person("Christine P'ng", role = "ctb"),
              person("Jeff Green", role = "ctb"),

--- a/NEWS
+++ b/NEWS
@@ -1,7 +1,11 @@
-BoutrosLab.plotting.general 7.1.2 2024-04-02
+
+
+
+BoutrosLab.plotting.general 7.1.2 2024-09-06
 
 UPDATE
 * Improved "sample.order" argument checking
+* Cleaned the C++ namespace by using `R_` versions of Free, Calloc, and Realloc
 
 --------------------------------------------------------------------------
 BoutrosLab.plotting.general 7.1.1 2024-01-08

--- a/src/ks.c
+++ b/src/ks.c
@@ -148,8 +148,8 @@ K(int n, double d)
    k = (int) (n * d) + 1;
    m = 2 * k - 1;
    h = k - n * d;
-   H = (double*) Calloc(m * m, double);
-   Q = (double*) Calloc(m * m, double);
+   H = (double*) R_Calloc(m * m, double);
+   Q = (double*) R_Calloc(m * m, double);
    for(i = 0; i < m; i++)
        for(j = 0; j < m; j++)
 	   if(i - j + 1 < 0)
@@ -177,8 +177,8 @@ K(int n, double d)
        }
    }
    s *= pow(10., eQ);
-   Free(H);
-   Free(Q);
+   R_Free(H);
+   R_Free(Q);
    return(s);
 }
 
@@ -215,7 +215,7 @@ m_power(double *A, int eA, double *V, int *eV, int m, int n)
 	return;
     }
     m_power(A, eA, V, eV, m, n / 2);
-    B = (double*) Calloc(m * m, double);
+    B = (double*) R_Calloc(m * m, double);
     m_multiply(V, V, B, m);
     eB = 2 * (*eV);
     if((n % 2) == 0) {
@@ -232,7 +232,7 @@ m_power(double *A, int eA, double *V, int *eV, int m, int n)
 	    V[i] = V[i] * 1e-140;
 	*eV += 140;
     }
-    Free(B);
+    R_Free(B);
 }
 
 /* Two-sided two-sample */

